src/ui/components/search/searchcomponent.js

```diff
@@ -138,6 +145,12 @@ export default class SearchComponent extends Component {
       if (this.queryEl) {
         this.queryEl.value = q;
       }
+      if (q === null) {
```

Comment:
What happens if somebody has a default initial search, makes a search, then clears the searchbar? I think we also just want the check to be
```js
if (!q && this._defaultInitialSearch)
```
it feels a little weird to care about things being null vs undefined

Reply:
I'm not sure I understand.

If somebody has a default initial search, makes a search, then clears the searchbar, the initial default search is re-done. I'm wary of the `!q` because it stops us from allowing empty search. If someone were to press enter on an empty searchbar we would want to search with an empty query as normal (note the `return`). We would not want to search, though, if the searchbar is emptied through the clear button or through a back nav, right?

Reply:
oh shoot my bad